xchange-coinbasepro/src/main/java/org/knowm/xchange/coinbasepro/CoinbaseProResilience.java

```diff
@@ -0,0 +1,48 @@
+package org.knowm.xchange.coinbasepro;
+
+import io.github.resilience4j.ratelimiter.RateLimiterConfig;
+import org.knowm.xchange.client.ResilienceRegistries;
+
+import java.time.Duration;
+
+public class CoinbaseProResilience {
+
+    public static final String ORDERS_PER_DAY_RATE_LIMITER = "ordersPerDay";
+
+    public static final String PRIVATE_PER_SECOND_RATE_LIMITER = "privateOrdersPerSecond";
+
+    public static final String PUBLIC_PER_SECOND_RATE_LIMITER = "publicOrdersPerSecond";
+
+    private CoinbaseProResilience() {
+    }
+
+    public static ResilienceRegistries createRegistries() {
+        ResilienceRegistries registries = new ResilienceRegistries();
+        registries
+                .rateLimiters()
+                .rateLimiter(
+                        PRIVATE_PER_SECOND_RATE_LIMITER,
+                        RateLimiterConfig.from(registries.rateLimiters().getDefaultConfig())
+                                .limitRefreshPeriod(Duration.ofSeconds(1))
+                                .limitForPeriod(5)
+                                .build());
+        registries
+                .rateLimiters()
+                .rateLimiter(
+                        PUBLIC_PER_SECOND_RATE_LIMITER,
+                        RateLimiterConfig.from(registries.rateLimiters().getDefaultConfig())
+                                .limitRefreshPeriod(Duration.ofSeconds(1))
+                                .limitForPeriod(3)
+                                .build());
+        registries
+                .rateLimiters()
+                .rateLimiter(
+                        ORDERS_PER_DAY_RATE_LIMITER,
+                        RateLimiterConfig.from(registries.rateLimiters().getDefaultConfig())
+                                .timeoutDuration(Duration.ZERO)
+                                .limitRefreshPeriod(Duration.ofDays(1))
+                                .limitForPeriod(500)
```

Comment:
How has this limit been decided?

Reply:
From the coinbase api docs:


PUBLIC ENDPOINTS
We throttle public endpoints by IP: 3 requests per second, up to 6 requests per second in bursts. Some endpoints may have custom rate limits.

PRIVATE ENDPOINTS
We throttle private endpoints by profile ID: 5 requests per second, up to 10 requests per second in bursts. Some endpoints may have custom rate limits.

Orders

You can place two types of orders: limit and market. Orders can only be placed if your account has sufficient funds. Each profile can have a maximum of 500 open orders on a product. Once reached, the profile will not be able to place any new orders until the total number of open orders is below 500. Once an order is placed, your account funds will be put on hold for the duration of the order. How much and which funds are put on hold depends on the order type and parameters specified. See the Holds details below.


Currently the rate limiter doesn't support the concept of having a max open order per product. I put it in there to see what others would say about that.

Reply:
Hey so thinking about this more I think we should get rid of this order limiter unless we can control it by product id. If you are trading a lot of symbols you could hit this limit and it would be an artificial limit.

Reply:
Ok, yah that makes sense. Will do.